frontend/pages/admin/OrgSettingsPage/cards/Advanced/Advanced.tsx

```diff
@@ -38,20 +60,35 @@ const Advanced = ({
     enableStartTLS,
     enableHostExpiry,
     hostExpiryWindow,
+    deleteActivities,
+    activityExpiryWindow,
     disableLiveQuery,
     disableScripts,
     disableQueryReports,
   } = formData;
 
-  const [formErrors, setFormErrors] = useState<IAppConfigFormErrors>({});
+  const [formErrors, setFormErrors] = useState<IAdvancedConfigFormErrors>({});
+
+  const activityExpiryWindowOptions = useMemo(
+    () =>
+      getCustomDropdownOptions(
+        ACTIVITY_EXPIRY_WINDOW_DROPDOWN_OPTIONS,
+        activityExpiryWindow,
+        // it's safe to assume that frequency is a number
+        (frequency: number | string) => `${frequency as number} days`
+      ),
+    // intentionally leave activityExpiryWindow out of the dependencies, so that the custom
+    // options are maintained even if the user changes the frequency in the UI
+    [deleteActivities]
+  );
 
-  const handleInputChange = ({ name, value }: IFormField) => {
+  const onInputChange = ({ name, value }: IFormField) => {
     setFormData({ ...formData, [name]: value });
   };
 
   useEffect(() => {
```

Comment:
next iteration we want to move towards validating with the `onInputChange` instead of after setting state